src/main/java/uk/nhs/tis/trainee/notifications/mapper/ProgrammeMembershipMapper.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright 2023 Crown Copyright (Health Education England)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+ * associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.nhs.tis.trainee.notifications.mapper;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Map;
+import org.mapstruct.Mapper;
+import org.mapstruct.MappingConstants.ComponentModel;
+import org.mapstruct.ReportingPolicy;
+import uk.nhs.tis.trainee.notifications.model.Curriculum;
+import uk.nhs.tis.trainee.notifications.model.ProgrammeMembership;
+
+/**
+ * A mapper to map between TIS Data and Programme Membership Data.
+ */
+@Mapper(componentModel = ComponentModel.SPRING, unmappedTargetPolicy = ReportingPolicy.IGNORE)
+public interface ProgrammeMembershipMapper {
+
+  /**
+   * Map a record data map to a ProgrammeMembership.
+   *
+   * @param recordData The map to convert.
+   * @return The mapped ProgrammeMembership.
+   */
+  default ProgrammeMembership toEntity(Map<String, String> recordData)
```

Comment:
This feels like losing out on the Jackson magic somehow, but I couldn't come up with anything less manual.

Reply:
You could define the defaults like `@Mapping(target = "tisId", source = "recordDate.tisId")` and use a post mapping method to handle the curriculum list.

Alternatively something like this _might_ work too
```java
@Mapping(`target="curricula", source = "data.curricula")
// Other mappings
ProgrammeMembership toEntity(Map<String, String> recordData);

default List<Curriculum> toCurricula(String) {
 // read value here
}
```